calculate_costs.py

```diff
@@ -0,0 +1,870 @@
+        if power_pv_nominal <= pv_nominal_power_max[0]:
+            feed_in_charge_pv = pv_remuneration[0]  # [ct/kWh]
+        elif power_pv_nominal <= pv_nominal_power_max[1]:
+            feed_in_charge_pv = pv_remuneration[1]  # [ct/kWh]
+        elif power_pv_nominal <= pv_nominal_power_max[2]:
+            feed_in_charge_pv = pv_remuneration[2]  # [ct/kWh]
+        else:
+            raise ValueError("Feed-in remuneration for PV is calculated only for nominal powers of "
+                             "up to 100 kWp")
+    # PV power plant not existing:
+    else:
+        feed_in_charge_pv = 0  # [ct/kWh]
+
+    # energy feed in by PV power plant:
+    energy_feed_in_pv_sim = sum(power_feed_in_list) * timestep_s / 3600  # [kWh]
+    energy_feed_in_pv_per_year = energy_feed_in_pv_sim * (duration_year_s / duration_sim_s)  # [kWh]
+
+    # costs for PV feed-in:
+    pv_feed_in_costs_sim = energy_feed_in_pv_sim * feed_in_charge_pv / 100  # [EUR]
+    pv_feed_in_costs_per_year = energy_feed_in_pv_per_year * feed_in_charge_pv / 100  # [EUR]
+
+    # COSTS FROM TAXES AND TOTAL COSTS:
+
+    # electricity tax:
+    electricity_tax = price_sheet["taxes"]["tax_on_electricity"]  # [ct/kWh]
+    electricity_tax_costs_sim = electricity_tax * energy_supply_sim / 100  # [EUR]
+    electricity_tax_costs_per_year = (electricity_tax_costs_sim
+                                      * (duration_year_s / duration_sim_s))  # [EUR]
+
+    # value added tax:
+    value_added_tax_percent = price_sheet["taxes"]["value_added_tax"]  # [%]
+    value_added_tax = value_added_tax_percent / 100  # [-]
+
+    # total costs without value added tax (commodity costs and capacity costs included):
+    costs_total_not_value_added_eur_sim = (
+        commodity_costs_eur_sim
+        + capacity_costs_eur_sim
+        + power_procurement_costs_sim
+        + additional_costs_sim
+        + levies_costs_total_sim
+        + concession_fee_costs_sim
+        + electricity_tax_costs_sim
+    )  # [EUR]
+    costs_total_not_value_added_eur_per_year = (costs_total_not_value_added_eur_sim
+                                                * (duration_year_s / duration_sim_s))  # [EUR]
+
+    # costs from value added tax:
+    value_added_tax_costs_sim = value_added_tax * costs_total_not_value_added_eur_sim
+    value_added_tax_costs_per_year = value_added_tax * costs_total_not_value_added_eur_per_year
+
+    # total costs with value added tax (commodity costs and capacity costs included):
+    costs_total_value_added_eur_sim = (costs_total_not_value_added_eur_sim
+                                       + value_added_tax_costs_sim)
+    costs_total_value_added_eur_per_year = (costs_total_not_value_added_eur_per_year
+                                            + value_added_tax_costs_per_year)
+
+    # MORE ADAPTATIONS REGARDING THE CUSTOMER TYPE:
+
+    capacity_or_basic_costs = "capacity costs"
+
+    if strategy == "greedy" or strategy == "balanced":
```

Comment:
This statement also appears in line 285

Reply:
I used the if statement two times because it is used for very different parts of the code ( 1) calculation of capacity and commodity costs, 2) writing costs into the simulation.json and adjust wordings according to strategy and customer type ). I think the code is easier to understand this way but I will discuss if we should reduce the if statements for efficiency reasons.

Reply:
Ok fine with me. Maybe ask Stefan if he agrees.